ports/servoshell/minibrowser.rs

```diff
@@ -314,4 +319,13 @@ impl Minibrowser {
             _ => false,
         }
     }
+
+    /// Updates the spinner from the given [WebViewManager] returning true iff a redraw is needed (always).
+    pub fn update_spinner_in_toolbar(
+        &mut self,
+        browser: &mut WebViewManager<dyn WindowPortsMethods>,
+    ) -> bool {
+        self.loading = browser.loading();
+        return true;
```

Comment:
Currently this always forces a `redraw`. I attempted to only redraw when `self.loading` changed like this:

```suggestion
        let need_redraw = self.loading != browser.loading();
        self.loading = browser.loading();
        return need_redraw;
```

But that resulted in some weird behavior where the spinner wasn't always being drawn. I'll investigate why this is the case but wanted to open the PR already to get feedback on the two questions in the other comments, I hope that's okay.

Reply:
This is resolved by the new commit, whch implements this without problems.